Description: Keep ABI for 18.08.3
 A method was removed from public symbols. For Debian we don't want to
 trigger an transition.
Author: Hefee <hefee@debian.org>
Author: Pino Toscano <pino@debian.org>
Origin: Debian
Forwarded: not-needed
Last-Update: 2020-11-22

--- a/src/prefs.cpp
+++ b/src/prefs.cpp
@@ -960,6 +960,22 @@ void Prefs::setResourceColor(const QStri
     d->mBaseConfig.setResourceColor(cal, color);
 }
 
+#define USE_RANDOM
+void Prefs::createNewColor(QColor &defColor, int seed)
+{
+#ifdef USE_RANDOM
+    Q_UNUSED(seed);
+    QColor col = QColor(QRandomGenerator::global()->bounded(256), QRandomGenerator::global()->bounded(256), QRandomGenerator::global()->bounded(256));
+    defColor = col;
+#else
+    int h, s, v;
+    defColor.getHsv(&h, &s, &v);
+    h = (seed % 12) * 30;
+    s -= s * static_cast<int>(((seed / 12) % 2) * 0.5);
+    defColor.setHsv(h, s, v);
+#endif
+}
+
 QColor Prefs::resourceColorKnown(const QString &cal) const
 {
     QColor color;
--- a/src/prefs.h
+++ b/src/prefs.h
@@ -218,6 +218,7 @@ public:
     Q_REQUIRED_RESULT int firstDayOfWeek() const;
 
     KConfig *config() const;
+    void createNewColor(QColor &defColor, int seed);
 
 private:
     class Private;
